mem.go

```diff
@@ -156,7 +156,7 @@ func ReadFromStream(r io.Reader, buf *bytes.Buffer) (*Segment, error) {
 	m := &MultiBuffer{make([]*Segment, segnum)}
 	for i := 0; i < segnum; i++ {
 		sz := int(binary.LittleEndian.Uint32(hdrv[4*i:])) * 8
-		m.Segments[i] = &Segment{m, datav[:sz], uint32(i), false}
+		m.Segments[i] = &Segment{m, append([]byte{}, datav[:sz]...), uint32(i), false}
```

Comment:
The [equivalent logic in v2](https://github.com/capnproto/go-capnproto2/blob/659aba4018b61e5f07f6b90ff2abc5b300baccea/mem.go#L409) just fixes the capacity of the slice to be the length, so that attempts to append to the slice will just allocate a new segment instead of clobbering other memory.

```suggestion
		m.Segments[i] = &Segment{m, datav[:sz:sz]), uint32(i), false}
```